activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -1198,8 +1198,13 @@ def build_statement_pool
         #
         # This is an internal hook to make possible connection adapters to build
         # custom result objects with connection-specific data.
-        def build_result(columns:, rows:, column_types: {})
-          ActiveRecord::Result.new(columns, rows, column_types)
+        def build_result(columns:, rows:, column_types: {}, raw_result: nil)
+          result_class.new(columns, rows, column_types, raw_result: raw_result)
```

Comment:
Little nit here 

Wondering why this is  `result_class.new(columns, rows, column_types, raw_result: raw_result)`
and not  `result_class.new(columns, rows, column_types, raw_result)`

Reply:
We are passing it as a keyword argument because this is what `initialize` of the `Result` class expects:
https://github.com/rails/rails/blob/1f773fa7ec14bb1dc1b986ad9306a58f8198e812/activerecord/lib/active_record/result.rb#L49

If the question is why `initialize` expects a keyword argument instead of a positional argument then I don't have a solid answer to this as keyword argument is my personal preference to make the signature more explicit. I have nothing against making it the last positional argument if we have reasons to do that

Reply:
Yes, that's what my question was about. Looks good !